src/navigation-commands.ts

```diff
@@ -13,19 +27,28 @@ interface NavigationCommand {
 *
 * @param obj The object to check.
 */
-export function isNavigationCommand(obj: any): boolean {
+export function isNavigationCommand(obj: any): obj is NavigationCommand {
   return obj && typeof obj.navigate === 'function';
 }
 
 /**
 * Used during the activation lifecycle to cause a redirect.
 */
-export class Redirect {
+export class Redirect implements NavigationCommand {
+
+  url: string;
+  /**@internal */
+  options: NavigationOptions;
+  /**@internal */
+  shouldContinueProcessing: boolean;
+
+  private router: Router;
+
   /**
    * @param url The URL fragment to use as the navigation destination.
    * @param options The navigation options.
    */
-  constructor(url: string, options: any = {}) {
+  constructor(url: string, options: Partial<NavigationOptions> = {}) {
```

Comment:
Why Partial?

Reply:
Here is what I need to work a bit more, `Redirect` is a commonly used class, and the existing typing for 2nd param is any. I need to make the new typing strong, while ensuring backward compat. I thought `Partial` would do the job but it seems I will need to do something else